cmd/validate/validate.go

```diff
@@ -134,16 +134,18 @@ func validateFile(fileSystem afero.Fs, file string) filewalker.Result {
 		if err == io.EOF {
 			break
 		}
-		if warcRecord.Type() == gowarc.Warcinfo {
-			warcInfoId = warcRecord.WarcHeader().GetId(gowarc.WarcRecordID)
-		}
+
 		result.IncrRecords()
 		result.IncrProcessed()
 		if err != nil {
 			result.AddError(fmt.Errorf("rec num: %d, offset: %d, cause: %w", result.Records(), currentOffset, err))
 			break
 		}
 
+		if warcRecord.Type() == gowarc.Warcinfo {
+			warcInfoId = warcRecord.WarcHeader().GetId(gowarc.WarcRecordID)
+		}
+
```

Comment:
I think this can actually be deleted, as it does not seem to do anything. I tested it in this branch: https://github.com/nlnwa/warchaeology/tree/temporary-test-branch
All tests passed.

Reply:
It's passed as an argument to the `Close` method of the `teeReader` where it is used as an input to the `CloseOutputFileHook`. 

https://github.com/nlnwa/warchaeology/blob/c57900bf00f5fa92b20a507f3da9d7b2d6deb39f/cmd/validate/validate.go#L122

Reply:
https://github.com/nlnwa/warchaeology/blob/ae347b299e81b94354dc9a5108b62d0c63f584e0/internal/hooks/hooks.go#L246

It looks ugly and it can be discussed if it should live on, but that discussion is not part of this PR.

Reply:
> I think this can actually be deleted, as it does not seem to do anything. I tested it in this branch: https://github.com/nlnwa/warchaeology/tree/temporary-test-branch All tests passed.

The test passes if you remove the block of code because the syntax is still valid, but `warcInfoId` will in that case always be `nil`.

Reply:
Discussed this IRL, and we agreed that the code needs updating, but it is outside the scope of this PR.